src/haz3lcore/dynamics/Evaluator.re

```diff
@@ -710,20 +710,11 @@ let rec evaluate: (ClosureEnvironment.t, DHExp.t) => m(EvaluatorResult.t) =
       };
 
     | ListLit(x1, x2, x3, ty, lst) =>
-      let+ evaluated_lst =
-        List.fold_right(
-          (ele, ele_list) => {
-            let* ele_list = ele_list;
-            let+ ele' = evaluate(env, ele);
-            switch (ele') {
-            | BoxedValue(ele') => [ele'] @ ele_list
-            | _ => [ele, ...ele_list]
-            };
-          },
-          lst,
-          return([]),
-        );
-      BoxedValue(ListLit(x1, x2, x3, ty, evaluated_lst));
+      let+ lst =
+        lst
+        |> List.map(el => evaluate(env, el) >>| EvaluatorResult.unbox)
+        |> sequence;
+      BoxedValue(ListLit(x1, x2, x3, ty, lst));
```

Comment:
If any element of the list is Indet, then the whole list should be indet. So not always a boxed value...

Reply:
Oops. Do we want to stop evaluation of elements when we hit an indet one, or just do everything regardless?

Reply:
we evaluate everything but its only a BoxedValue if all of the arguments are BoxedValues (similar to tuples)